doc/debugger.rst

```diff
@@ -1388,6 +1402,27 @@ reestablish the link).
 ``msg`` is an optional string elaborating on the reason for the detach.  It may
 or may not be present depending on the nature of detachment.
 
+AppNotify notification (0x07)
+-----------------------------
+
+Format::
+
+    NFY <int: 0x07> [<tval>]* EOM
+
+Example::
+
+    NFY 7 "DebugPrint" "Everything is going according to plan!" EOM
+
+This is a custom notification message whose meaning and semantics depend on the
+application.
+
+AppNotify messages are used for direct communication between the debug client
+and debug target over the Duktape debug protocol. Both the meaning of a custom
+message and the dvalues it contains are entirely up to the implementation and
+depending on the needs of the application, need not be supported at all.
+
```

Comment:
I found that I actually prefer to use an integer - it's easier to work with in C because you can't `switch` on strings.  For clients written in a higher-level language though (probably the usual case), I agree that a string convention makes the most sense.  I'll find a way to work it in.